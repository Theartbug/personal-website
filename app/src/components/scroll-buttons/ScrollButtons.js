import React, { useEffect } from 'react';
import { Events } from 'react-scroll';
import { setCurrentSectionByButtons, setButtonScroll } from '../appContext/actions.js';
import { useMyContext } from '../appContext/AppContext.js';
import UpAngleIcon from 'react-icons/lib/fa/caret-up';
import DownAngleIcon from 'react-icons/lib/fa/caret-down';
import './scroll-buttons.css';

export default function ScrollButtons(props) {
  // const { setButtonScroll, currentSection, setCurrentSectionByButtons } = props;
  const { currentSection: { currentSection }, dispatch } = useMyContext();
  
  useEffect(setScrollListener, []);

  function setScrollListener() {
    //so the intersection observer does not interfere with react-scroll, must keep track in state
    Events.scrollEvent.register('end', () => dispatch(setButtonScroll(false)));

    // return the removal of the scrollEvent for hook
    return () => Events.scrollEvent.remove('end');
  }

  function handleDownClick() {
  //if the currentSection position is at the end and the direction wants to go further, dont let it.
    if(currentSection === 5) return;
    const next = currentSection + 1;
    dispatch(setCurrentSectionByButtons(next));
  };

  function handleUpClick() {
  //if the currentSection position is at the beginning the direction wants to go further, dont let it.
    if(currentSection === 0) return;
    const prev = currentSection - 1;
    dispatch(setCurrentSectionByButtons(prev));
  };

  return (
    <div className="scroll-buttons">
      <button 
        className="up reset-button" 
        onClick={ handleUpClick }>
        <UpAngleIcon/>
      </button>
      <button 
        className="down reset-button" 
        onClick={ handleDownClick }>
        <DownAngleIcon/>
      </button>
    </div>
  );
}