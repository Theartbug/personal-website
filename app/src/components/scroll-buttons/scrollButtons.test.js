import React from 'react';
import { shallow } from 'enzyme';
import { ScrollButtons } from './ScrollButtons';

describe('ScrollButton render tests', () => {
  const setCurrentSectionByButtons = jest.fn();
  const component = shallow(<ScrollButtons setCurrentSectionByButtons={setCurrentSectionByButtons}/>);
  
  it('should render correctly', () => {
    
    expect(component).toMatchSnapshot();
  });
  
  it('should change when hovering', () => {
    component.find('.up').simulate('mouseover');

    expect(component).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    component.find('.up').simulate('click');
    component.find('.down').simulate('click');

    expect(setCurrentSectionByButtons).toHaveBeenCalled();
    expect(setCurrentSectionByButtons).toHaveBeenCalled();
  });
  
});