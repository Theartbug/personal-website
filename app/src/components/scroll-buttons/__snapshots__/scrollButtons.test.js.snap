// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButton render tests should change when hovering 1`] = `
<div
  className="scroll-buttons"
>
  <button
    className="up reset-button"
    onClick={[Function]}
  >
    <FaCaretUp />
  </button>
  <button
    className="down reset-button"
    onClick={[Function]}
  >
    <FaCaretDown />
  </button>
</div>
`;

exports[`ScrollButton render tests should render correctly 1`] = `
<div
  className="scroll-buttons"
>
  <button
    className="up reset-button"
    onClick={[Function]}
  >
    <FaCaretUp />
  </button>
  <button
    className="down reset-button"
    onClick={[Function]}
  >
    <FaCaretDown />
  </button>
</div>
`;
